src/App.tsx

```diff
@@ -1,24 +1,59 @@
 import React from 'react';
+
+import { Clock } from './components/Clock';
+
 import './App.scss';
 
-const App: React.FC = () => {
-  const timerId: NodeJS.Timer = setInterval(() => {
-    const date: Date = new Date();
-
-    // eslint-disable-next-line
-    console.log(date.toLocaleTimeString());
-  }, 1000);
-
-  return (
-    <div className="App">
-      <h1>React clock</h1>
-      <p>
-        Current time:
-        {' '}
-        {/* Print the time here instead of DevTools */}
-      </p>
-    </div>
-  );
+type State = {
+  isClockVisible: boolean;
+  clockName: number;
 };
 
+class App extends React.Component<{}, State> {
+  state: State = {
+    isClockVisible: true,
+    clockName: Math.floor(Math.random() * 100),
+  };
+
+  showClock = () => {
+    this.setState({
+      isClockVisible: true,
+    });
+  };
+
+  hideClock = () => {
+    this.setState({
+      isClockVisible: false,
+    });
+  };
+
+  renameClock = () => {
+    this.setState({
+      clockName: Math.floor(Math.random() * 100),
+    });
+  };
+
+  render() {
+    const { isClockVisible, clockName } = this.state;
+
+    return (
+      <div className="App">
+        {isClockVisible && <Clock clockName={clockName} />}
+
+        <div className="buttons">
+          <button type="button" onClick={this.showClock}>
+            Show Clock
+          </button>
+          <button type="button" onClick={this.hideClock}>
+            Hide Clock
+          </button>
```

Comment:
- I can still "hide" the clock when it's already hidden, same with "show clock"
![image](https://user-images.githubusercontent.com/37161315/143418565-2383d4e8-7ce4-4dc9-b5e2-600dbcd1f800.png)